src/pykeen/datasets/inductive/base.py

```diff
@@ -0,0 +1,316 @@
+# -*- coding: utf-8 -*-
+
+"""Utility classes for constructing inductive datasets."""
+
+from __future__ import annotations
+
+import logging
+import pathlib
+from typing import Any, Mapping, Optional, Union
+
+from pystow.utils import download, name_from_url
+from tabulate import tabulate
+
+from ...constants import PYKEEN_DATASETS
+from ...triples import CoreTriplesFactory, TriplesFactory
+
+__all__ = [
+    "InductiveDataset",
+    "LazyInductiveDataset",
+    "DisjointInductivePathDataset",
+    "UnpackedRemoteDisjointInductiveDataset",
+]
+
+logger = logging.getLogger(__name__)
+
+
+class InductiveDataset:
+    """Contains transductive train and inductive inference/validation/test datasets."""
+
+    #: A factory wrapping the training triples
+    transductive_training: CoreTriplesFactory
+    #: A factory wrapping the inductive inference triples that MIGHT or MIGHT NOT
+    # share indices with the transductive training
```

Comment:
If they share IDs, how is this seen from the model?

Reply:
With InductiveNodePiece this is ok - entities in inductive inference will contain the same hashes in the same order as in the transductive training. The tricky part is to make sure IDs are contiguous, eg, `0...K` in train and `K+1 ... N` at inference